liquibase-core/src/main/java/liquibase/lockservice/StandardLockService.java

```diff
@@ -94,72 +97,96 @@ public void init() throws DatabaseException {
         boolean createdTable = false;
         Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc",  database);
 
-        if (!hasDatabaseChangeLogLockTable()) {
+        int maxIterations = 10;
+        for (int i = 0; i < maxIterations; i++) {
             try {
-                executor.comment("Create Database Lock Table");
-                executor.execute(new CreateDatabaseChangeLogLockTableStatement());
-                database.commit();
-                Scope.getCurrentScope().getLog(getClass()).fine(
-                        "Created database lock table with name: " +
-                                database.escapeTableName(
+                if (!hasDatabaseChangeLogLockTable(true)) {
+                    try {
+                        executor.comment("Create Database Lock Table");
+                        executor.execute(new CreateDatabaseChangeLogLockTableStatement());
+                        database.commit();
+                        Scope.getCurrentScope().getLog(getClass()).fine(
+                                "Created database lock table with name: " +
+                                        database.escapeTableName(
+                                                database.getLiquibaseCatalogName(),
+                                                database.getLiquibaseSchemaName(),
+                                                database.getDatabaseChangeLogLockTableName()
+                                        )
+                        );
+                        this.hasDatabaseChangeLogLockTable = true;
+                        createdTable = true;
+                        hasDatabaseChangeLogLockTable = true;
+                    } catch (DatabaseException e) {
+                        if ((e.getMessage() != null) && e.getMessage().contains("exist")) {
+                            //hit a race condition where the table got created by another node.
+                            Scope.getCurrentScope().getLog(getClass()).fine("Database lock table already appears to exist " +
+                                    "due to exception: " + e.getMessage() + ". Continuing on");
+                            // If another node already created the table, then we need to rollback this current transaction,
+                            // otherwise servers like Postgres will not allow continued use of the same connection, failing with
+                            // a message like "current transaction is aborted, commands ignored until end of transaction block"
+                            if (database instanceof PostgresDatabase) {
```

Comment:
The postgresql rollback probably needs to be in the overall catch logic, not just only in this special case.

Maybe should do a rollback regardless and not just with postgresql?

Reply:
The answer to both questions is yes, based on my testing.